File: uploads/dd1b5db0-6c1c-45c1-8810-73072968a72a/source/test_project/src/com/example/service/UserService.java
package com.example.service;

import com.example.model.User;
import java.util.ArrayList;
import java.util.List;

public class UserService {
    private List<User> userDatabase;
    private int nextId = 1;

    public UserService() {
        this.userDatabase = new ArrayList<>();
    }

    public User createUser(String name, String email) {
        User newUser = new User(nextId++, name, email);
        userDatabase.add(newUser);
        System.out.println("Created new user: " + newUser.getName());
        return newUser;
    }

    public User findUserById(int id) {
        for (User user : userDatabase) {
            if (user.getId() == id) {
                return user;
            }
        }
        return null; // User not found
    }

    public void printAllUsers() {
        System.out.println("--- All Users ---");
        if (userDatabase.isEmpty()) {
            System.out.println("No users in the database.");
        } else {
            for (User user : userDatabase) {
                System.out.println(user.toString());
            }
        }
        System.out.println("-----------------");
    }
}

File: outputs/7df5c56e-e1c2-450b-ba3d-2e3a39c12326/source/test_project/src/com/example/service/UserService.py
# Unit test result: ✅ PASSED
from typing import Optional
from com.example.model.User import User

class UserService:
    userDatabase: list['User']
    nextId: int = 1

    def __init__(self):
        self.userDatabase = []

    def create_user(self, name: str, email: str) -> 'User':
        new_user = User(self.nextId, name, email)
        self.nextId += 1
        self.userDatabase.append(new_user)
        print(f'Created new user: {new_user.get_name()}')
        return new_user

    def find_user_by_id(self, id: int) -> Optional['User']:
        for user in self.userDatabase:
            if user.get_id() == id:
                return user
        return None

    def print_all_users(self) -> None:
        print('--- All Users ---')
        if not self.userDatabase:
            print('No users in the database.')
        else:
            for user in self.userDatabase:
                print(str(user))
        print('-----------------')